### PR TITLE
Added FAQArticle comments

### LINKED ISSUE #88: Add FAQArticleComment mutation
Wrap this endpoint here https://gitlab.skypicker.com/django-team/knowledgebase#article-comment
to be able to send user feedback as comments for each FAQArticle.
Definition:

```
mutation FAQArticleComment(
   type: "confusing" | "doesntAnswer" | "dontLike" | "notAccurate" | "other", 
   comment: string) : FAQArticle`
```

### PR DESCRIPTION
**Pull request by git-toni**
Closes #88.

_Note: `{ "message": "yummy indeed" }` is the actual successful response, also a sequel to `{ "message": "yummy" }`_

### CODE REVIEW

**Comment by jaroslav-kubicek** [comment]
> Note: { "message": "yummy indeed" } is the actual successful response, also a sequel to { "message": "yummy" }

the best comment ever 😀 💯

File: src/FAQ/mutaions/FAQArticleComment.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+const successfulResponse = { message: 'yummy indeed' };
+export default {
+  type: FAQArticle,
+  args: {
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'ID of FAQ article to receive vote.',
+    },
+    language: {
+      type: LanguageInput,
```
Comment (jaroslav-kubicek) [inline src/FAQ/mutaions/FAQArticleComment.js]: deprecated, use `Accept-Language` header

note to self: update Readme finally 💥

Comment (mrtnzlml) [reply src/FAQ/mutaions/FAQArticleComment.js]: > note to self: update Readme finally 💥

@jaroslav-kubicek Can you please send a PR?

**Comment by git-toni** [comment]
@jaroslav-kubicek I checked with Michal. There's no comment limit at backend :+1:

File: src/FAQ/mutaions/FAQArticleComment.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'ID of FAQ article to receive vote.',
+    },
+    type: {
+      type: FAQCommentType,
+      description:
+        'Comments can be of one of these types: confusing|doesntAnswer|dontLike|notAccurate|other',
+    },
+    comment: {
+      type: GraphQLString,
```
Comment (jaroslav-kubicek) [inline src/FAQ/mutaions/FAQArticleComment.js]: should be nonullable - without comment it doesn't make much sense

File: src/FAQ/mutaions/FAQArticleComment.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+const successfulResponse = { message: 'yummy indeed' };
+export default {
+  type: FAQArticle,
+  args: {
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'ID of FAQ article to receive vote.',
+    },
+    type: {
```
Comment (jaroslav-kubicek) [inline src/FAQ/mutaions/FAQArticleComment.js]: type should be also non-null or with some default

File: src/FAQ/types/inputs/FAQCommentType.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
+// @flow
+
+import { GraphQLEnumType } from 'graphql';
+
+export default new GraphQLEnumType({
+  name: 'FAQCommentType',
+  values: {
+    confusing: { value: 'confusing' },
```
Comment (jaroslav-kubicek) [inline src/FAQ/types/inputs/FAQCommentType.js]: I would advise to use UPPERCASE for keys `confusing: { value: 'confusing' }` => `CONFUSING: { value: 'confusing' }`

File: src/FAQ/mutaions/FAQArticleComment.js (hunk 3 above)
Comment (git-toni) [reply src/FAQ/mutaions/FAQArticleComment.js]: Agree about the non-nullable. Disagree about the default, I think it should specifically match a type.

File: src/FAQ/mutaions/FAQArticleComment.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+      type: new GraphQLNonNull(FAQCommentType),
+      description:
+        'Comments can be of one of these types: confusing|doesntAnswer|dontLike|notAccurate|other',
```
Comment (michalsanger) [inline src/FAQ/mutaions/FAQArticleComment.js]: I wouldn't write the options here, will quickly get out of sync with changes in `FAQCommentType`

File: src/FAQ/mutaions/FAQArticleComment.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
+// @flow
```
Comment (michalsanger) [inline src/FAQ/mutaions/FAQArticleComment.js]: Could you fix the typo in path? `mutaions` should be `mutations`

File: src/FAQ/mutaions/__tests__/FAQArticleComment.test.js [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    expect(
+      await graphql(commentMutation, {
+        articleId: globalId,
+        comment: 'hola',
+        type: 'OTHER',
+      }),
+    ).toMatchSnapshot();
+  });
+  it('should return WRONG comment type', async () => {
+    RestApiMock.onPost(
```
Comment (michalsanger) [inline src/FAQ/mutaions/__tests__/FAQArticleComment.test.js]: Is the mock needed? Invalid type should be checked before any API is being called

File: src/FAQ/mutaions/FAQArticleComment.js (hunk 6 above)
Comment (git-toni) [reply src/FAQ/mutaions/FAQArticleComment.js]: :man_facepalming:

File: src/FAQ/mutations/FAQArticleComment.js [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+  type: FAQArticle,
+  args: {
+    id: {
+      type: new GraphQLNonNull(GraphQLID),
+      description: 'ID of FAQ article to receive vote.',
+    },
+    type: {
+      type: new GraphQLNonNull(FAQCommentType),
+      description:
+        'Value indicating the comment type. Check kiwibase documentation.',
```
Comment (michalsanger) [inline src/FAQ/mutations/FAQArticleComment.js]: "Check kiwibase documentation" is useless for non-kiwi users of our GraphQL. Just the "Value indicating the comment type" is OK, values and meaning should be clear from inspecting the `FAQCommentType`

File: scripts/graphql-schema-snapshot.json [hunk 9]
```diff
@@ -7457,6 +7457,61 @@
...
+                    "kind": "SCALAR",
+                    "name": "ID",
+                    "ofType": null
+                  }
+                },
+                "defaultValue": null
+              },
+              {
+                "name": "type",
+                "description": "Comments can be of one of these types: confusing|doesntAnswer|dontLike|notAccurate|other",
```
Comment (michalsanger) [inline scripts/graphql-schema-snapshot.json]: Looks like this isn't refreshed

File: src/RootMutation.js [hunk 10]
```diff
@@ -3,11 +3,13 @@
...
 
 import Login from './identity/mutations/Login';
+import FAQArticleComment from './FAQ/mutations/FAQArticleComment';
 
 export default new GraphQLObjectType({
   name: 'RootMutation',
   description: 'Root Mutation.',
   fields: {
     login: Login,
+    FAQArticleComment,
```
Comment (michalsanger) [inline src/RootMutation.js]: I missed this one. Could you name the mutation `addFAQArticleComment` please?

**Pull request merged**